apps/meteor/client/views/invite/hooks/useValidateInviteQuery.ts

```diff
@@ -0,0 +1,53 @@
+import { useEndpoint, useSessionDispatch, useSetting, useToastMessageDispatch, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+
+import { useInviteTokenMutation } from './useInviteTokenMutation';
+
+export const useValidateInviteQuery = (userId: string, token: string) => {
+	const t = useTranslation();
+
+	const registrationForm = useSetting('Accounts_RegistrationForm');
+
+	const setLoginDefaultState = useSessionDispatch('loginDefaultState');
+	const dispatchToastMessage = useToastMessageDispatch();
+	const mutateInviteToken = useInviteTokenMutation();
+
+	const handleValidateInviteToken = useEndpoint('POST', '/v1/validateInviteToken');
+
+	return useQuery(
+		['invite', token],
+		async () => {
+			if (!token) {
+				return false;
+			}
```

Comment:
But then I would have two different returns, no? One would be the useQuery object with all the isLoading, isSuccess params and the other only a boolean. Or are you suggesting that I should build a "handmade" useQuery object and return it? If this is the case I believe it would add unneeded complexity to this code and possibly break the tanstack query logic flow.

Reply:
You didn't typed your return, so you can do that, even returning a toastError... This prevents calling the endpoint when you don't have the necessary information